packages/eslint-plugin/tests/eslint-rules/no-redeclare.test.ts

```diff
@@ -22,7 +60,211 @@ type Foo = 1;
 function foo({ bar }: { bar: string }) {
     console.log(bar);
 }
+    `,
+    `
+type AST<T extends ParserOptions> = TSESTree.Program &
+  (T['range'] extends true ? { range: [number, number] } : {}) &
+  (T['tokens'] extends true ? { tokens: TSESTree.Token[] } : {}) &
+  (T['comment'] extends true ? { comments: TSESTree.Comment[] } : {});
+interface ParseAndGenerateServicesResult<T extends ParserOptions> {
+  ast: AST<T>;
+  services: ParserServices;
+}
+    `,
+    `
+function A<T>() {}
+interface B<T> {}
+type C<T> = Array<T>
+class D<T> {}
     `
   ],
-  invalid: []
+  invalid: [
+    {
+      code: 'var a = 3; var a = 10;',
+      parserOptions: { ecmaVersion: 6 },
+      errors: [
+        {
+          message: "'a' is already defined.",
+          type: AST_NODE_TYPES.Identifier
+        } as any
```

Comment:
ugh, we should consider PRing the base eslint to migrate their rules to messageId

Reply:
yup we should :)